src/iceberg/util/checked_cast_internal.h

```diff
@@ -0,0 +1,61 @@
+/*
```

Comment:
Should we install this header by removing the internal suffix? I don't see it being used in this PR, but in #94, it's included in visit_type.h, which is meant to be installed for type manipulation. If we don't install this header, it might lead to compilation issues.

Reply:
I was using it for the visitor but not it is not required. Let me remove it for now. I saw that you have added that in your PR.